servicetest/testframework/lib.py

```diff
@@ -32,9 +32,105 @@
 import Queue
 import subprocess
 import dbus.mainloop.glib
+import os.path
 from gi.repository import GObject
 
 
+class ConfigFile(object):
+    """ Represents the configuration file for SoftwareContainer
+        All keys and values have to be strings.
+
+        The format of a config file is as follows:
+
+        [Group1]
+        item1 = value1
+        item2 = value2
+
+        [Group2]
+        item1 = value1
+        item2 = value2
+
+        Initialization is done as follows:
+
+        config = ConfigFile("/where/to/save/me", {
+            "Group1": {
+                "item1": "value1",
+                "item2": "value2"
+            },
+
+            "Group2": {
+                "item1": "value1",
+                "item2": "value2"
+            }
+        })
+    """
+
+    def __init__(self, location, full_config):
+        """ Create a config object from a given dict of dicts.
+        """
+        self.__filename = location
+        self.__groups = full_config
+
+    def filename(self):
```

Comment:
This actually returns the full path including filename right? A name change might be appropriate in that case.